dell/src/main/java/org/apache/iceberg/dell/ecs/EcsTableOperations.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.dell.ecs;
+
+import java.util.Map;
+import org.apache.iceberg.BaseMetastoreTableOperations;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+
+public class EcsTableOperations extends BaseMetastoreTableOperations {
+
+  public static final String ICEBERG_METADATA_LOCATION = "iceberg_metadata_location";
+
+  private final String tableName;
+  private final FileIO fileIO;
+  private final EcsCatalog catalog;
+  private final EcsURI tableObject;
+
+  /**
+   * Cached E-Tag for CAS commit
+   *
+   * @see #doRefresh() when reset this field
+   * @see #doCommit(TableMetadata, TableMetadata) when use this field
+   */
+  private String eTag;
+
+  public EcsTableOperations(String tableName, EcsURI tableObject, FileIO fileIO, EcsCatalog catalog) {
+    this.tableName = tableName;
+    this.tableObject = tableObject;
+    this.fileIO = fileIO;
+    this.catalog = catalog;
+  }
+
+  @Override
+  protected String tableName() {
+    return tableName;
+  }
+
+  @Override
+  public FileIO io() {
+    return fileIO;
+  }
+
+  @Override
+  protected void doRefresh() {
+    String metadataLocation;
+    if (!catalog.objectMetadata(tableObject).isPresent()) {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchTableException("Metadata object %s is absent", tableObject);
```

Comment:
As there are some operations in `EcsCatalog` that we don't guarantee the atomicity, so there may remain some metadata files while the object meta data is absent ?  I think we need a more clear message to indicate the case.